frontend/src/Onboarding/TransferableCreditScreen.tsx

```diff
@@ -0,0 +1,182 @@
+      ...props.selectedTransferableExams,
+      props.transferableExam,
+    ];
+    props.setSelectedTransferableExams(newSelectedTransferableExams);
+  };
+
+  const removeCourseFromSelected = () => {
+    const newSelectedTransferableExams: Array<
+      TransferableExam
+    > = props.selectedTransferableExams.filter(
+      (transferableExam: TransferableExam) =>
+        transferableExam.name !== props.transferableExam.name
+    );
+    props.setSelectedTransferableExams(newSelectedTransferableExams);
+  };
+
+  const onChecked = (e: any): void => {
+    const checked = e.target.checked;
+    if (checked) {
+      addCourseToSelected();
+    } else {
+      removeCourseFromSelected();
+    }
+  };
+
+  return (
+    <SelectableCourse
+      key={`${props.transferableExam.type}-${props.transferableExam.name}`}
+      onChange={onChecked}
+      courseText={props.transferableExam.name}
+    />
+  );
+};
+
+const TransferableExamGroupComponent: React.FC<
+  TransferableExamGroupComponentProps
+> = props => {
+  return (
+    <div>
+      <TitleText>{props.transferableExamGroup.name}</TitleText>
+      {props.transferableExamGroup.transferableExams.map(
+        (transferableExam: TransferableExam) => (
+          <TransferableExamComponent
+            transferableExam={transferableExam}
+            selectedTransferableExams={props.selectedTransferableExams}
+            setSelectedTransferableExams={props.setSelectedTransferableExams}
+            key={`${transferableExam.type}-${transferableExam.name}`}
+          />
+        )
+      )}
+    </div>
+  );
+};
+
+const TransferableExamGroupsComponent: React.FC<
+  TransferableExamGroupsComponentProps
+> = props => {
+  return (
+    <div>
+      {props.transferableExamGroups.map(
+        (transferableExamGroup: TransferableExamGroup) => (
+          <TransferableExamGroupComponent
+            transferableExamGroup={transferableExamGroup}
+            selectedTransferableExams={props.selectedTransferableExams}
+            setSelectedTransferableExams={props.setSelectedTransferableExams}
+            key={`${props.keyPrefix}-${transferableExamGroup.name}`}
+          />
+        )
+      )}
+    </div>
+  );
+};
+
+const TransferableCreditScreen: React.FC = () => {
+  const dispatch = useDispatch();
+  const [selectedTransferableExams, setSelectedTransferableExams] = useState<
+    Array<TransferableExam>
+  >([]);
+
+  const onSubmit = (): void => {
+    dispatch(setExamCredits(selectedTransferableExams));
+  };
+
+  // indicates if the user came from login button on welcome page
+  const location = useLocation();
+  const { fromOnBoardingGuest } = (location.state as any) || {
+    fromOnBoardingGuest: false,
+  };
+
+  return (
+    <OnboardingSelectionTemplate
+      screen={3}
+      mainTitleText={"Select any exams you took for AP or IB credit:"}
+      onSubmit={onSubmit}
+      to={fromOnBoardingGuest ? "/home" : "/signup"}
```

Comment:
Brought this over from the completed courses screen since this is now the current last page within the onboarding process